datafusion/physical-optimizer/src/pruning.rs

```diff
@@ -258,9 +258,7 @@ pub trait PruningStatistics {
 ///
 /// * Rows that evaluate to `false` are not returned (“filtered out” or “pruned” or “skipped”).
 ///
-/// * Rows that evaluate to `NULL` are **NOT** returned (also “filtered out”).
-///   Note: *this treatment of `NULL` is **DIFFERENT** than how `NULL` is treated
-///   in the rewritten predicate described below.*
+/// * No rows should evaluate to `null`, even if statistics are missing (the statistics are themselves `null`).
```

Comment:
I have much more docstrings to update, including examples, but I will hold off on that until I get feedback on the rest of the PR.